pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -167,7 +167,8 @@ func (r *ReconcileZookeeperCluster) Reconcile(request reconcile.Request) (reconc
 func zookeeperClusterFresherThanSts(instance *zookeeperv1beta1.ZookeeperCluster, sts *appsv1.StatefulSet) bool {
```

Comment:
Thinking more I think we should refactor this into a "compareResourceVersion(zk, sts)" and return -1,0 or 1 

When zk version is lt sts version we should keep the current logic.
When zk version is gt sts version then sts should inherit zk version
When equal do nothing

Reply:
It  makes sense